lib/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,167 @@
+#include "accelerometer.hpp"
+
+namespace hyped::sensors {
+
+Accelerometer::Accelerometer(const std::uint8_t mux_address,
+                             const std::uint8_t channel,
+                             io::I2c &i2c,
+                             core::ILogger &log)
+    : mux_address_(mux_address),
+      channel_(channel),
+      log_(log),
+      i2c_(i2c)
+{
+}
+
+Accelerometer::~Accelerometer()
+{
+}
+
+std::uint8_t Accelerometer::getChannel()
+{
+  return channel_;
+}
+
+std::optional<std::int16_t> Accelerometer::getRawAccelerationX()
```

Comment:
This setup seems very verbose? Unless I'm missing something, the only real difference between getting the raw accelerations for x, y, and z is the register address you read (and the error message but that doesn't really affect functionality)? Therefore would it not make sense to pass that in as a function argument for a single function (e.g. `getRawAccelerationAxis(register_address)`) or similar. Then you would only have to write one function?

Reply:
No, the user should definitely not have to pass in the correct register address.

Having a private function like that seems fine, though.

Maybe there's a better approach here? Not that it's too important but one idea I'd have is making an enum for X, Y, and Z and then passing in that value? The function can then convert the enum into register values and do the work? This means the caller isn't required to pass magic register addresses. There are problems with this; perhaps you have better ideas?

Reply:
Maybe not a register address then but some way of selecting which axis to read from. Defining 3 separate functions with only trivial changes between them seems quite pointless?

Reply:
maybe like a `getRawAccelerationHelper` (naming open to suggestions ofc) taking in two enum values for low and high byte (+1 for axis to use in log messages?)

Reply:
drop the `Helper` and make it private